crates/state/src/gs.rs

```diff
@@ -0,0 +1,153 @@
+#![warn(unused)]
+
+use crate::storage::{Fingerprint, Storage};
+use crate::StorageResult as Result;
+use svm_types::{Address, BytesPrimitive, Layer, TemplateAddr};
+
+pub struct GlobalState {
+    storage: Storage,
+}
+
+impl GlobalState {
+    pub async fn new(sqlite_uri: &str) -> Self {
+        Self {
+            storage: Storage::new(sqlite_uri).await.unwrap(),
+        }
+    }
+
+    pub async fn in_memory() -> Self {
+        Self {
+            storage: Storage::in_memory().await.unwrap(),
+        }
+    }
+
+    pub async fn account_balance(&self, account_addr: &Address) -> Result<Option<u64>> {
+        let key = format!("accounts:{}:balance", account_addr.to_string());
```

Comment:
might be better to use shorter encoding here.
also I think that the Account data should be mapped by a single key